pkg/controller/gittrack/git_creds.go

```diff
@@ -17,18 +17,71 @@ limitations under the License.
 package gittrack
 
 import (
+	"context"
 	"fmt"
 	"strings"
 
 	farosv1alpha1 "github.com/pusher/faros/pkg/apis/faros/v1alpha1"
 	gitstore "github.com/pusher/git-store"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 type gitCredentials struct {
 	secret         []byte
 	credentialType farosv1alpha1.GitCredentialType
 }
 
+// fetchGitCredentials creates git credentials data from a given deployKey secret reference
+func (r *ReconcileGitTrack) fetchGitCredentials(gt farosv1alpha1.GitTrackInterface) (*gitCredentials, error) {
+	deployKey := gt.GetSpec().DeployKey
+
+	// Check if the deployKey is empty, do nothing if it is
+	emptyKey := farosv1alpha1.GitTrackDeployKey{}
+	if deployKey == emptyKey {
+		return nil, nil
+	}
+	// Check the deployKey fields are both non-empty
+	if deployKey.SecretName == "" || deployKey.Key == "" {
+		return nil, fmt.Errorf("if using a deploy key, both SecretName and Key must be set")
```

Comment:
```suggestion
		return nil, fmt.Errorf("both SecretName and Key must be set when using DeployKey")
```

Reply:
This was a copy pasta but happy to fix up